crates/csharp/src/lib.rs

```diff
@@ -1175,24 +1175,22 @@ impl<'a> wit_bindgen_core::InterfaceGenerator<'a> for InterfaceGenerator<'a> {
                 } else {
                     ""
                 };
-                format!(
-                    "public static readonly {name} {flag_name} = new {name}(({ty}) (1{suffix} << {i}));"
-                )
+                format!("{flag_name} = 1{suffix} << {i},")
             })
             .collect::<Vec<_>>()
             .join("\n");
 
+        let enum_type = if matches!(flags.repr(), FlagsRepr::U32(2)) {
```

Comment:
Shouldn't this be set to the corresponding type based on the number of flags as in the specification as seen here? https://github.com/WebAssembly/component-model/blob/main/design/mvp/CanonicalABI.md#alignment

Reply:
Thanks, updated to use `ushort` and `byte`.